### DISCUSSION TITLE
Very many files... a couple UI tweaks?

### DISCUSSION THREAD

**Discussion by DonEdwards** [geany/geany#3105] [Ideas]
I'm currently working on something where I often open around 360 HTML files simultaneously. This causes a couple issues:

1) My title bar is showing the full (rather long) path, and has to abbreviate the filename (of the file that currently has focus) to make room. I really need to know the complete filename, but the path is (in my case) the same for all files - and therefore the preferred thing to leave parts out of.

2) The tab title bars also abbreviate filenames. Not much can be done about that, but it makes going to any specific file difficult. Could there be a dropdown for this purpose? Note that this also needs to show complete filenames - and must have a scrollbar when the list gets long.

Thanks.

**Comment by elextr** [comment]
Somebody has made the rather arbitary (and hardcoded, boo) decision that the filename should be truncated to 30 bytes, but the path should be unlimited. Geany has no way of knowing how long to make the window titlebar AFAIK since it depends on window size and what decorations the WM uses, but sensible improvements in the current situation will probably be accepted. [Edit: also it is a common use-case to have multiple copies of a file with the same name but in different directories open, so then the path is what is important, so mindlessly hacking the path is also open to question]

In the meantime, there is a right click dropdown, but with 360 files its probably not much use.

There is also the document tab on the sidebar which is more useful for large numbers of files.

**Reply by eht16** [reply]
> Somebody has made the rather arbitary (and hardcoded, boo) decision that the filename should be truncated to 30 bytes, but the path 

Somebody = me.
The initial implementation was in f1eb8dc22 and was requested in https://sourceforge.net/p/geany/bugs/298/.
It might be a good idea to make the width configurable and by setting it to 0 to disable the feature completely. But as always, it needs someone to do it.

> In the meantime, there is a right click dropdown, but with 360 files its probably not much use.

The Addons plugin also provides a toolbar icon to quickly show a list of open files though it's pretty much the same as the right click dropdown on the tar bar.
